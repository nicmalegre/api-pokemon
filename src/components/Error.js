import React from 'react';
import Snackbar from '@material-ui/core/Snackbar';
import MuiAlert from '@material-ui/lab/Alert';
import { makeStyles } from '@material-ui/core/styles';

function Alert(props) {
  return <MuiAlert elevation={6} variant="filled" {...props} />;
}

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    '& > * + *': {
      marginTop: theme.spacing(2),
    },
  },
}));

export const Error = ({isError}) => {
  const classes = useStyles();
  const [open, setOpen] = React.useState(isError);

  const handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      return;
    }
    setOpen(false);
  };

  return (
    <div className={classes.root}>
      <Snackbar 
      open={open} 
      autoHideDuration={6000} 
      onClose={handleClose}>
        <Alert severity="error" onClose={handleClose}>Ha ocurrido un error. Por favor vuelva a intentarlo.</Alert>
      </Snackbar>
    </div>
  );
}

