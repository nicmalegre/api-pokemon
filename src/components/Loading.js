import React from 'react'

//Import from Material-UI
import Backdrop from '@material-ui/core/Backdrop';
import CircularProgress from '@material-ui/core/CircularProgress';
import { makeStyles } from '@material-ui/core/styles';


/* STYLE FOR BACKDROP */
const useStyles = makeStyles((theme) => ({
    backdrop: {
    zIndex: theme.zIndex.drawer + 1,
    color: '#fff',
    },
}));


export const Loading = ({loading}) => {
    const classes = useStyles(); //This is for the material-ui styles 

    return(
        <>
        {/* This component will display when the API is loading */}
            <Backdrop className={classes.backdrop} open={loading}>
                <CircularProgress color="inherit" />
            </Backdrop>
        {/* End Backdrop */}
        </>
    )


}

