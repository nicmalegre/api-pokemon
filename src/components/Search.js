import React from 'react'

//Import from Material-ui
import { FormControl, Select} from '@material-ui/core/';
import MenuItem from '@material-ui/core/MenuItem';
import TextField from '@material-ui/core/TextField';
import Grid from '@material-ui/core/Grid';


export const Search = ({limit, count,setLimit, onSearch, setSearch}) => {

    //Save the input state in the search state
    const handleChangeInput = (event) =>{
        setSearch(event.target.value)
    }

    //When the select change, we save the new limit
    const handleChangeSelect = (event) => {
        setLimit(event.target.value);
    };
    
    return (
        <Grid item xs={12} style={{marginBottom: 40}}>
          <p style={{fontSize: '24px', fontWeight: 'lighter'}}>El que quiere Pokemon, que los busque.</p>
          <Grid container justify="center">
              <TextField 
              id="outlined-basic" 
              variant="outlined"  
              size='small' 
              style={{width: '70%', marginRight: 20}} 
              label='Ingrese el nombre a buscar'
              onChange={handleChangeInput}
              />
            <FormControl variant="outlined"  size='small'>
              
              <Select
                labelId="demo-simple-select-outlined-label"
                id="demo-simple-select-outlined"
                value={limit}
                onChange={handleChangeSelect}
                size="small"
              > 
                <MenuItem value={10}>10</MenuItem>
                <MenuItem value={20}>20</MenuItem>
                <MenuItem value={30}>30</MenuItem>
                <MenuItem value={50}>50</MenuItem>
                <MenuItem value={75}>75</MenuItem>
                <MenuItem value={100}>100</MenuItem>
                <MenuItem value={200}>200</MenuItem>
                
                {(limit >= 200) && (
                  <MenuItem value={limit}>{limit}</MenuItem>
                )}
              </Select>
            </FormControl>
          </Grid>
          
          
        </Grid>
    )


}
