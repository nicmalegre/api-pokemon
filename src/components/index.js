export {Title} from './Title'
export {Search} from './Search'
export {PokemonCard} from './PokemonCard'
export {Loading} from './Loading'
export {Error} from './Error'