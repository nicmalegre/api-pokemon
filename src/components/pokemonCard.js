import React, { useContext, useEffect, useState } from 'react';

/* IMPORT FROM MATERIAL-UI */
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { green } from '@material-ui/core/colors';


/* STYLES FOR THE CARD IMPORT FROM MATERIAL-UI */
const useStyles = makeStyles({
    root: {
      maxWidth: 320,
      minWidth:300,
      display: 'inline-block',
      margin: 15,
      padding: 15,
      
    },
    media: {
      height: 140,
    },
  });


const PokemonCard = (props) => {
    
    const classes = useStyles(); //We need to declare this in order to use 

    return ( 
        <>
        <Card className={classes.root}>
            <div style={{textAlign: 'center'}}>
                <img height='120px' src={props.pokemon.data.sprites.front_default} alt=''/>
                <img height='120px' src={props.pokemon.data.sprites.back_default} alt=''/>
            </div>
            <p key={props.pokemon.data.name}><b>Name:</b> {props.pokemon.data.name}</p>
            <p key={props.pokemon.data.height}><b>Height:</b> {props.pokemon.data.height}</p>
            <p key={props.pokemon.data.weight}><b>Weight:</b> {props.pokemon.data.weight}</p>
            <p key={props.pokemon.data.base_experience}><b>Base Experience:</b> {props.pokemon.data.base_experience}</p>
            <p><b>Abilities: </b>
                <ul>
                    {props.pokemon.data.abilities.map(a =>{ 
                        return (
                            <li key={a.ability.name}>{a.ability.name}</li>
                        )
                    })}
                </ul>
            </p> 
        </Card> 
        </>

  );
}
 
export default PokemonCard;