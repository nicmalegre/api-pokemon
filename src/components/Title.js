import React from 'react';

import {Container, AppBar, Toolbar, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

import Avatar from '@material-ui/core/Avatar';

import src from '../assets/img/nicolasalegre.jpg'



//ESTO SE UTILIZA PARA DARLE ESTILO A LA NAVBAR
const useStyles = makeStyles((theme) => ({
    root: {
      flexGrow: 1,
    },
    menuButton: {
      marginRight: theme.spacing(2),
    },
    title: {
      flexGrow: 1,
    },
    avatar: {
        display: 'flex',
        '& > *': {
            margin: theme.spacing(1),
        },
    },
    [theme.breakpoints.down('sm')]:{
      avatar: {display:'none'}
    },
  }));

const Title = () => {
  const classes = useStyles();  

  return ( 
      <>
    <div className={classes.root}>
        <AppBar position="static" style={{backgroundColor:"#3f51b5"}}>
          <Container justify='center'>
            <Toolbar style={{padding:0}}>
                <h1 style={{fontSize: '40px'}}>Pokemon Finder</h1>
                <Typography variant="h6" className={classes.title}>
                </Typography>
                
                <div>
                    <label><b>Author:</b></label><br/>
                    <label id='a-name'>Nicol√°s Alegre</label><br/>
                    <a id='a-href' href='https://github.com/nicmalegre/api-pokemon'>Link to GitHub</a>
                </div>
                <div className={classes.avatar}>
                    <Avatar id="avatar" style={{height: '70px', width: '70px'}} alt="Nicolas Alegre" src={src} />
                </div>
            </Toolbar>
          </Container>
        </AppBar>
      </div> 
      </>
  )
}

export default(Title);
