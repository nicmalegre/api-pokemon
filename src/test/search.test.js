import React from 'react';
import renderer from 'react-test-renderer';
import {Search} from '../components/Search';
  
  it('renders Search component correctly', () => {
    const tree = renderer
      .create(<Search />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
