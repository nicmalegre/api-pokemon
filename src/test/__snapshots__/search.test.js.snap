// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Search component correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  style={
    Object {
      "marginBottom": 40,
    }
  }
>
  <p
    style={
      Object {
        "fontSize": "24px",
        "fontWeight": "lighter",
      }
    }
  >
    El que quiere Pokemon, que los busque.
  </p>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      className="MuiFormControl-root MuiTextField-root"
      style={
        Object {
          "marginRight": 20,
          "width": "70%",
        }
      }
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
        data-shrink={false}
        htmlFor="outlined-basic"
        id="outlined-basic-label"
      >
        Ingrese el nombre a buscar
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          disabled={false}
          id="outlined-basic"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              Ingrese el nombre a buscar
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        onClick={[Function]}
        size="small"
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="demo-simple-select-outlined-label demo-simple-select-outlined"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          id="demo-simple-select-outlined"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          aria-hidden={true}
          className="MuiSelect-nativeInput"
          onAnimationStart={[Function]}
          onChange={[Function]}
          required={false}
          tabIndex={-1}
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-2"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
